hollow/src/main/java/com/netflix/hollow/core/index/HollowHashIndex.java

```diff
@@ -146,10 +145,10 @@ private boolean matchIsEqual(FixedLengthElementArray matchHashTable, long hashBu
             int hashOrdinal = (int)matchHashTable.getElementValue(hashBucketBit + hashState.getOffsetPerTraverserField()[field.getBaseIteratorFieldIdx()], hashState.getBitsPerTraverserField()[field.getBaseIteratorFieldIdx()]) - 1;
 
             HollowTypeReadState readState = field.getBaseDataAccess();
-            int fieldPath[] = field.getSchemaFieldPositionPath();
+            int[] fieldPath = field.getSchemaFieldPositionPath();
 
             if(fieldPath.length == 0) {
-                if(hashOrdinal != ((Integer)query[i]).intValue())
+                if (query[i].equals(hashOrdinal))
```

Comment:
needs a `!`

Reply:
Indeed! Thank you a lot !!
Tests didn't help here unfortunately